### PR TITLE
Enhance CMake files

### PR DESCRIPTION
**Pull request by kovdan01**
- Add install rule for sqlpp-mysql cmake package that provides sqlpp11::mysql exported target
- Use canonical cmake way of handling dependencies via `find_package` instead of custom `DATE_INCLUDE_DIR` and `SQLPP11_INCLUDE_DIR` variables.
- Do not use legacy `include_directories` commands.
- Enhance FindMySQL.cmake module:
  - Create exported target MySQL::MySQL.
  - Fix behavior when MariaDB files are located in mysql subdirectory instead of mariadb (see Arch Linux mariadb-libs package: https://archlinux.org/packages/extra/x86_64/mariadb-libs/)
- Run JsonTests with MariaDB as well as with MySQL.

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -14,34 +14,36 @@ matrix:
...
   - cd date
-  - git checkout tags/v2.4
-  - cd ..
-  - git clone https://github.com/rbock/sqlpp11
+  - git checkout tags/v3.0.1
+  - mkdir build && mkdir prefix && cd build
+  - cmake -DCMAKE_INSTALL_PREFIX="%cd%\..\prefix" -DBUILD_TZ_LIB=OFF ..
+  - cmake --build . --target install
+  - cd ..\..
+  - git clone https://github.com/kovdan01/sqlpp11.git
```
Comment (rbock) [inline .appveyor.yml]: I guess we'll have to wait for you PR in sqlpp11 to land?

File: .travis.yml [hunk 2]
```diff
@@ -38,30 +38,53 @@ install:
...
+  - git checkout tags/v3.0.1
+  - mkdir build && mkdir prefix && cd build
+  - >
+    cmake \
+      -D CMAKE_INSTALL_PREFIX=`pwd`/../prefix \
+      -D BUILD_TZ_LIB=OFF \
+      ..
+  - cmake --build . --target install
+  - cd ../..
+  - git clone https://github.com/kovdan01/sqlpp11.git
```
Comment (rbock) [inline .travis.yml]: same here?

File: .appveyor.yml (hunk 1 above)
Comment (kovdan01) [reply .appveyor.yml]: Ah, sorry, forgot about that. Temporarily cloned my own repository when used "Visual Studio 2013" worker image on appveyor - it only supports CMake 3.12.2, while sqlpp11 library requires CMake 3.14 or higher. Now "Visual Studio 2019" worker image with CMake 3.20.2 is used, so there is no need in using my fork. Changed to using 0.60 tag from original rbock/sqlpp11.

File: .travis.yml (hunk 2 above)
Comment (kovdan01) [reply .travis.yml]: See answer to the previous comment

**Comment by rbock** [comment]
Thanks for this very thorough change and the quick update. Looks good to me.

**Comment by kovdan01** [comment]
@rbock You're welcome, thank you for your library!

By the way, just have realized that README.md is now a bit outdated
- `DATE_INCLUDE_DIR` and `SQLPP11_INCLUDE_DIR` are not used anymore. 
- That is not a problem anymore - in current CI build the library links with pre-installed MySQL C API. I guess that this should be removed from readme as for now.
  > Remark regarding MSVC builds on appveyor: As of now, I can neither to link against the provided mysql-connector-c library nor build it from scratch. I am currently compiling the mariadb client and link against that one. Build is fine, tests fail. Help is appreciated.

Should I create a PR with readme update, or you can do it by yourself? 

And one more thing: it would be cool if you could tag a new release after these changes, so readme would correspond to the latest release. If you decide to do so – please don’t forget to change library version in top-level CMakeLists.txt

**Comment by rbock** [comment]
I would appreciate an other PR and would create a release afterwards.

Regarding the release number in the CMakeLists.txt: I know I will forget this at some point. I wonder if some is someone has already written a pre push hook script to prevent this?

**Comment by kovdan01** [comment]
Created PR with README.md update: #74 

Regarding a pre push hook script: I'll look for a good solution for this. A possible prototype is implemented in #66, but I don't like it because CMakeLists.txt depends on git (while a popular user scenario is to download release archive instead of cloning the repo; git might be even not installed in the system, especially on Windows).

I'll let you know when I find a good (from my perspective) solution.

**Comment by rbock** [comment]
Right, the approach in #66 is not ideal. Thanks for looking for something better.

**Pull request merged**